pkg/devicemapper/devmapper_wrapper_dynamic.go

```diff
@@ -2,5 +2,135 @@
 
 package devicemapper
 
-// #cgo pkg-config: devmapper
+/*
+#cgo pkg-config: devmapper
+#cgo LDFLAGS: -ldl
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <libdevmapper.h>
+
+// Yes, I know this looks scary. In order to be able to fill our own internal
```

Comment:
Do you think there would be a need to have a build-flag to make this feature "opt-in"?

Reply:
Perhaps. However it should be noted that this warning:

```
// It is also important that this structure be _larger_ than the dm_info that
// libdevmapper expected. Otherwise libdm might try to write to memory it
// shouldn't (they don't have a "known size" API).
```

Applies to the existing codebase as well. If `libdm` adds extra fields and Docker isn't rebuilt, then it is also possible for `libdm` to overwrite memory it shouldn't (which could cause a segfault). The scary-looking part of this code actually adds extra padding which should future-proof this.